pkg/reconciler/pullsubscription/pullsubscription.go

```diff
@@ -172,19 +206,40 @@ func (c *Reconciler) reconcile(ctx context.Context, source *v1alpha1.PullSubscri
 		source.Status.MarkSink(sinkURI)
 	}
 
-	subID := resources.GenerateSubName(source)
+	source.Status.SubscriptionID = resources.GenerateSubscriptionName(source)
 
-	cont, err := c.ensureSubscription(ctx, source, subID)
-	if err != nil {
-		logger.Error("Unable to ensure subscription", zap.Error(err))
+	state, err := c.EnsureSubscription(ctx, source, source.Spec.Project, source.Spec.Topic, source.Status.SubscriptionID)
+	switch state {
+	case pubsub.OpsGetFailedState:
+		logger.Error("Failed to get subscription ops job.", zap.Any("state", state), zap.Error(err))
 		return err
-	}
-	if !cont {
-		logger.Info("Waiting for Job.")
+
+	case pubsub.OpsCreatedState:
+		// If we created a job to make a subscription, then add the finalizer and update the status.
+		addFinalizer(source)
+		source.Status.MarkSubscriptionOperation("Creating",
+			"Created Job to create Subscription %q.",
+			source.Status.SubscriptionID)
```

Comment:
Is there a possible race here where the SubscriptionID gets stored in `status`, the subscribe job gets created, and then the resource is deleted, and a delete job is created while the create job is still running?

At which point, does the job lookup in `getJob` work correctly, or does it return a random job?

Reply:
oh interesting. and uug... I will make a bug for this one.

Reply:
https://github.com/GoogleCloudPlatform/cloud-run-events/issues/29